dozer-types/src/models/connection.rs

```diff
@@ -16,30 +20,175 @@ pub struct Connection {
 
 #[derive(Serialize, Deserialize, Eq, PartialEq, Clone, ::prost::Message, Hash)]
 pub struct PostgresConfig {
-    #[prost(string, tag = "1")]
+    #[prost(string, optional, tag = "1")]
+    pub user: Option<String>,
+    #[prost(string, optional, tag = "2")]
+    pub password: Option<String>,
+    #[prost(string, optional, tag = "3")]
+    pub host: Option<String>,
+    #[prost(uint32, optional, tag = "4")]
+    pub port: Option<u32>,
+    #[prost(string, optional, tag = "5")]
+    pub database: Option<String>,
+    #[prost(string, optional, tag = "6")]
+    pub sslmode: Option<String>,
+    #[prost(string, optional, tag = "7")]
+    pub connection_url: Option<String>,
+}
+
+#[derive(Eq, PartialEq, Clone, Debug, Hash)]
+pub struct PostgresConfigReplenished {
     pub user: String,
-    #[prost(string, tag = "2")]
     pub password: String,
-    #[prost(string, tag = "3")]
     pub host: String,
-    #[prost(uint32, tag = "4")]
     pub port: u32,
-    #[prost(string, tag = "5")]
     pub database: String,
+    pub ssl_mode: SslMode,
 }
 
-impl PostgresConfig {
+impl PostgresConfigReplenished {
     pub fn convert_to_table(&self) -> Table {
         table!(
-            ["user", self.user.as_str()],
+            ["user", self.user],
             ["password", "*************"],
             ["host", self.host],
             ["port", self.port],
-            ["database", self.database]
+            ["database", self.database],
+            ["sslmode", format!("{:?}", self.ssl_mode)]
         )
     }
 }
 
+impl PostgresConfig {
+    pub fn replenish(&self) -> PostgresConfigReplenished {
```

Comment:
Should we return an error instead of `unwrap`? With helpful error messages like `user` must be present if `connection_url` is not present.

Reply:
Can we do `.expect()` with helpful error message instead?

Reply:
That's kind of inconsistent with what we're doing now. Why we need to use `expect`?

Reply:
to get rid of extra layer of Result checking with .unwrap()

Reply:
I can think of two downsides of getting rid of `Result`:

1. Inconsistency with error handling elsewhere.
2. Make it difficult if upper layer wants to handle the error originated in `replenish`.

What's the benefit here?

Reply:
converting the function to return with `Result`